src/backend/common/CommandBuffer.cpp

```diff
@@ -358,21 +443,44 @@ namespace backend {
                     }
                     break;
 
+                case Command::EndRenderPass:
+                    {
+                        iterator.NextCommand<EndRenderPassCmd>();
+                        if (currentRenderPass == nullptr) {
+                            device->HandleError("No render pass is currently active");
+                            return false;
+                        }
+                        currentRenderPass = nullptr;
+                        currentFramebuffer = nullptr;
+                        lastPipeline = nullptr;
+                        aspects.reset(VALIDATION_ASPECT_RENDER_PIPELINE);
+                        bindgroupsSet.reset();
+                        aspects.reset(VALIDATION_ASPECT_BINDGROUPS);
+                        inputsSet.reset();
+                        aspects.reset(VALIDATION_ASPECT_VERTEX_BUFFERS);
+                        aspects.reset(VALIDATION_ASPECT_INDEX_BUFFER);
+                    }
+                    break;
+
                 case Command::SetPipeline:
                     {
                         SetPipelineCmd* cmd = iterator.NextCommand<SetPipelineCmd>();
                         PipelineBase* pipeline = cmd->pipeline.Get();
                         PipelineLayoutBase* layout = pipeline->GetLayout();
 
                         if (pipeline->IsCompute()) {
+                            if (currentRenderPass != nullptr) {
+                                device->HandleError("Can't use a compute pipeline while a render pass is active");
+                                return false;
+                            }
                             aspects.set(VALIDATION_ASPECT_COMPUTE_PIPELINE);
-                            aspects.reset(VALIDATION_ASPECT_RENDER_PIPELINE);
                         } else {
+                            if (currentRenderPass == nullptr) {
+                                device->HandleError("A render pass must be active when a render pipeline is set");
+                                return false;
+                            }
                             aspects.set(VALIDATION_ASPECT_RENDER_PIPELINE);
-                            aspects.reset(VALIDATION_ASPECT_COMPUTE_PIPELINE);
                         }
-                        aspects.reset(VALIDATION_ASPECT_BINDGROUPS);
```

Comment:
This dirties the validation of vertex buffers and bindgroups so that we can perform it again at draw time. It should be kept.